### ISSUE TITLE
Consider making `#[fastly::main]` comply with `clippy::pedantic`

### ISSUE THREAD

**Issue by matanlurey** [fastly/fastly-rust#8]
Most examples of getting started with Rust and Fastly introduce the following pattern:

```rs
#[fastly::main]
fn main(req: Request) -> Result<Response, Error> { /* ... */ }
```

For folks who enable [`clippy::pedantic`](https://doc.rust-lang.org/clippy/lints.html#pedantic), you'll get at least 3 lint warnings:

- `clippy::unnecessary_wraps`

   Most code samples never return an `Error`, making `Result<Response, Error>` unnecessary.
   
   The macro could allow a return value of pure `Response`.

- `clippy::needless_pass_by_value`

  The request object (`req: Request`) does not always need ownership.
  
  The macro could allow a parameter of `&Request`.

- `clippy::inline_always`

  This is a false-positive, so probably the right thing to do is just tell users using pedantic to ignore it.

It's possible to work around these issues by following the instructions to bootstrap without the macro, or by using `#[allow(clippy::lint_warning)]` but figured I'd share the feedback anyway and see if any of these were intentional or easy to improve upon or if it's worth documenting using stricter lints with the Fastly Rust API.

Thanks!

**Comment by harmony7** [comment]
Hey, thanks for this report. This issue doesn't actually belong in this project, as this is for the generated fastly-rust crate meant for consuming the Fastly API in Rust applications outside of Compute.

Your issue is regarding the Rust SDK for Compute, whose source code isn't currently open source, so there is no GitHub repository to file tickets.  Instead, please make a post in the Fastly Community Forum under the Rust SDK category: https://community.fastly.com/c/compute/compute-rust-sdk/19
